### PR TITLE
enable scope middleware with generic res body.

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Feature

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview

Enable `Scope::wrap` and `Scope::wrap_fn` accepting middlware that transform response body without boxing.
The final output `Scope` must still have `BoxBody`. This can be achieved with `Compat` or another middleware does the same job without recieving any other middleware type

### CODE REVIEW

File: src/scope.rs [hunk 1]
```diff
@@ -304,23 +307,24 @@ where
     /// ServiceResponse.
```
Comment (aliemjay) [inline src/scope.rs]: Docs on wrap and wrap_fn need an update

**Review by aliemjay — CHANGES_REQUESTED** [review]
I think we need the same thing for Resource.

I'll do it in a followup PR if you like.

**Comment by aliemjay** [comment]
semver-major?

**Comment by robjtede** [comment]
> semver-major

ye adding a type param to wrap makes it breaking, strictly, though i doubt anyone would be using it in a way that it would _actually_ break

**Review by robjtede — APPROVED** [review]

**Pull request merged**